### PR TITLE
Fix for #339

### PR DESCRIPTION
**Pull request by Jesuszilla**
This should fix the error I was having, though it's hard to tell now that I'm getting a different error for a different file (it can't find a .NET property that actually exists on a .NET object), so I can't really test it.

### CODE REVIEW

File: src/Metatables.cs [hunk 1]
```diff
@@ -1037,38 +1037,47 @@ bool TrySetMember(LuaState luaState, ProxyType targetType, object target, Bindin
...
-                catch (Exception e)
-                {
-                    ThrowError(luaState, e);
-                }
-
+                SetMemberCache(targetType, fieldName, member);
                 return true;
             }
-            if (member.MemberType == MemberTypes.Property)
+            else if (member is MemberInfo)
```
Comment (viniciusjarina) [inline src/Metatables.cs]: Maybe we can drop the `else` if we early return the first `if` as well.

File: src/Metatables.cs [hunk 2]
```diff
@@ -1037,38 +1037,47 @@ bool TrySetMember(LuaState luaState, ProxyType targetType, object target, Bindin
                 member = members[0];
                 SetMemberCache(targetType, fieldName, member);
             }
-
-            if (member.MemberType == MemberTypes.Field)
+            else if (member is LuaNativeFunction)
```
[the remark this answers is not in the document]
Comment (Jesuszilla) [reply src/Metatables.cs]: That's the type I get when I assign a table value to a function.

File: src/Metatables.cs (hunk 1 above)
Comment (Jesuszilla) [reply src/Metatables.cs]: Ah, I missed that return. Will fix.

**Comment by Jesuszilla** [comment]
@viniciusjarina Please review. Don't know what the holdup is on the other two tests but it's been days so I assume they are being run on local machines.

**Comment by viniciusjarina** [comment]
So apparently there are tests failing yet `InvalidArrayIndex` 
Also, one of the reasons we added  the PushInvalidMethod was due slow performance calling invalid methods https://github.com/NLua/NLua/issues/315 
So I understand this patch is fixing an issue that you have, but what is the issue? Why isn't possible to write a test for it?
I would be happy to merge it ,if we don't regress, and if we have tests covering the case for this addition.

Thank you

**Pull request closed without merging**